system/HTTP/CURLRequest.php

```diff
@@ -489,6 +489,9 @@ protected function applyRequestHeaders(array $curl_options = []): array
 	{
 		$headers = $this->getHeaders();
 
+		// Unset 'Host' header, CURL will determinate it automatically from URL.
+		unset($headers['Host']);
```

Comment:
I don't think we can do this. There are many different scenarios where `Host` header may be necessary, i.e when you call internal IP on the server.

Reply:
> I don't think we can do this. There are many different scenarios where `Host` header may be necessary, i.e when you call internal IP on the server.

True, we can put here check if is set custom header for Host, if not it will be unset, if yes it will use set from options. Because now in every request you have to define Host, it will never determinate it automatically what is redundant to write it to each request.

Reply:
When you are grabbing service without options it will set base_uri like a host in example case it is 'example.com' or for local development server, it will be 'localhost or so... Case is that if you are trying to access any server you need to specify host each time. It is not necessary since CURL will set Host based on URL. Another thing is that it is not mentioned in the guide that you need set host to make valid request for another server. Simply it is redundant and it confuses people why there is 404 error when you set full valid URL but CI will use localhost all the time.

For example, this will produce error.
$client = \Config\Services::curlrequest();
$response = $client->get('https://google.com');

This is directly from the guide:
$client = \Config\Services::curlrequest();
$response = $client->request('GET', 'https://api.github.com/user', [
        'auth' => ['user', 'pass']
]);

It will produce 404 error as well.

Reply:
@xbotkaj - thanks for the details, now I get it.

I think the solution would be a change in `Message` class for `populateHeaders()` method. We should skip `HTTP_HOST` in foreach loop:

```php
if ($key === 'HTTP_HOST')
{
    continue;
}
```

Reply:
@michalsn yes it can be done by your way, but it would be good to check if by someone since it changes Message and it is deeper than CURLrequest class that is extend of Request and that is extend of Message class.

For fix just in CURL we could use something like this in `applyRequestHeaders()` :
```
if (empty($this->headers))
{
	$headers = $this->getHeaders();
	unset($headers['Host']);
}
else
{
	$headers = $this->getHeaders();
}		
```